apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/NamespaceService.java

```
@@ -204,56 +208,59 @@ public NamespaceBO findPublicNamespaceForAssociatedNamespace(Env env, String app
+            //Get the item list asynchronously
+            Future<List<ItemDTO>> itemsAsync = namespaceBoAsyncService.getItemsAsync(appId, env, clusterName, namespaceName);
+            //This method takes the most time, so make it in the main thread
+            List<ItemDTO> deletedItems = itemService.findDeletedItems(appId, env, clusterName, namespaceName);
+            Map<String, ItemDTO> deletedItemsMap = new HashMap<>();
+            //Convert list to map
+            if (!CollectionUtils.isEmpty(deletedItems)) {
+                deletedItems.forEach(itemDTO -> deletedItemsMap.put(itemDTO.getKey(), itemDTO));
+            }
+            List<ItemDTO> list = itemsAsync.get();
+            if (!CollectionUtils.isEmpty(list)) {
+                items.addAll(list);
+            }
+            Map<String, String> releaseAsync = latestReleaseAsync.get();
+            if (Objects.nonNull(releaseAsync)) {
+                releaseItems.putAll(releaseAsync);
+            }
+            // Convert itemDTO' s list to ItemBO 's list
+            List<ItemBO> itemBOs = items
+                    .parallelStream()
+                    .map(itemDTO -> transformItem2BO(itemDTO, releaseItems))
+                    .collect(Collectors.toList());
+            // Count the number of modified items
+            long modifiedItemCnt = itemBOs
+                    .parallelStream()
+                    .filter(ItemBO::isModified)
+                    .count();
+            // Parse deleted items
+            List<ItemBO> deletedItemBos = parseDeletedItems(items, releaseItems, deletedItemsMap);
+            itemBOs.addAll(deletedItemBos);
+            namespaceBO.setItems(itemBOs);
+            // The number of items to be modified
+            namespaceBO.setItemModifiedCnt((int) modifiedItemCnt + deletedItemBos.size());
+        } catch (InterruptedException | ExecutionException e) {
+            logger.error("transformNamespace2BO error namespace:{}, env:{},err:{}",
```

logger.error doesn't need to include the err in the message.

---

Why do we swallow the exception here?

---

抛出去上层处理吗